### PR TITLE
Add functionality to build models inside R

### PR DESCRIPTION
**Pull request by WardBrian**
This addresses part of #100 and is more or less a straight port of the Python functionality. 

I'm not at all a confident R programmer, so I'm going to ask if @jgabry or @rok-cesnovar could kindly spare some time to look this over.

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
If you can walk me through this, would be happy to review it. We can schedule on Slack.

**Review by rok-cesnovar — APPROVED** [review]
Brian walked me through the changes and it all makes sense to me.

File: R/R/compile.R [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    res_attrs <- attributes(res)
+    if ("status" %in% names(res_attrs) && res_attrs$status != 0) {
+        stop(paste0("Compilation failed with error code ", res_attrs$status, "\noutput:\n",
+            paste(res, collapse = "\n")))
+    }
+
+    return(output)
+}
+
+windows_path_setup <- function() {
```
Comment (roualdes) [inline R/R/compile.R:101]: I missed it.  Where is this function used?

**Review by roualdes — APPROVED** [review]
Thanks to you both, Brian and Rok.  There's some nits in here, but overall this looks great.  Much appreciated.

File: R/R/compile.R (hunk 1 above)
Comment (WardBrian) [reply R/R/compile.R:101]: It's called in the constructor before we load the shared object (same as Python, looks like Julia calls it at startup which is not ideal and I will take a look at)

File: R/R/compile.R [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+
+#' Run BridgeStan's Makefile on a `.stan` file, creating the `.so`
+#' used by the StanModel class.
+#' This function assumes that the path to BridgeStan is valid.
+#' This can be set with `set_bridgestan_path`.
+#'
+#' @param stan_file A path to a Stan model file.
+#' @param stanc_arg A list of arguments to pass to stanc3.
```
[the remark this answers is not in the document]
Comment (WardBrian) [reply R/R/compile.R:63]: I think vector is the correct terminology - blame my lack of R familiarity

File: R/R/compile.R [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        })
+    }
+
+    return(path)
+}
+
+
+#' Run BridgeStan's Makefile on a `.stan` file, creating the `.so`
+#' used by the StanModel class.
+#' This function assumes that the path to BridgeStan is valid.
```
[the remark this answers is not in the document]
Comment (WardBrian) [reply R/R/compile.R:59]: Good question. I guess the sucessful execution of the function assumes the path is valid? But we should probably say "This function _checks_..."

**Review by roualdes — APPROVED** [review]
Looks good to me.  Thanks much.

**Pull request merged**